src/catalog.cpp

```diff
@@ -597,6 +597,9 @@ void Catalog::SetFileName(const wxString& fn)
 {
     wxFileName f(fn);
     f.Normalize(wxPATH_NORM_DOTS | wxPATH_NORM_ABSOLUTE);
+    auto name = f.GetName();
+    name.ToLong(&m_crowdinProjectId);
+    name.AfterFirst('_').ToLong(&m_crowdinFileId);
```

Comment:
Addressed in a711235a66f13ccf01d2f70967ec1a4b9522e203 both fragile parsing and put together all Crowdin related logic/behaviour in `Catalog` class to single `IsFromCrowdin()` while still remained `m_crowdinFileId` and `m_crowdinProjectId` since they are involved in `Save/Sync` `.po` files downloaded from Crowdin UI in browser - just to store file and project IDs appropriately on first `Save/Sync` and to do not need to retrieve  them (making few expensive Crowdin API calls) on each further `Save/Sync` (until file is closed of course). I hope that's what you expected or please feel free to `Unresolve` this conversation otherwsie